File: Copilot/4.Patrón de Diseño Singleton/javascript/singleton.js
const express = require('express');
const app = express();

app.get('/saludo', (req, res) => {
    const nombre = req.query.nombre || 'amigo';
    res.send(`Hola, ${nombre}!`);
});

app.listen(3000, () => {
    console.log('Servidor en http://localhost:3000');
});

File: Copilot/4.Patrón de Diseño Singleton/golang/singleton.go
package main

import (
    "fmt"
    "net/http"
)

func saludoHandler(w http.ResponseWriter, r *http.Request) {
    nombre := r.URL.Query().Get("nombre")
    if nombre == "" {
        nombre = "amigo"
    }
    fmt.Fprintf(w, "Hola, %s!", nombre)
}

func main() {
    http.HandleFunc("/saludo", saludoHandler)
    fmt.Println("Servidor en http://localhost:8080")
    http.ListenAndServe(":8080", nil)
}
